### DISCUSSION TITLE
Take action on rollback/abort event

### DISCUSSION THREAD

**Discussion by pshrm** [argoproj/argo-rollouts#3153] [Q&A]
I'm fairly new to argo world and exploring argo rollout to integrate in to our Kubernetes cluster for canary deployments of services. I have an external dependency (outside of cluster) which has to be in sync with the service deployed in the cluster. I would like to know how can I listen to a rollback/abort event from argo rollout and perform a rollback (my own custom logic) in external dependency. One way could be using Notifications/Services/Webhook (I would need to run a server)but is there a another way to do it? 
Thanks!

**Comment by kostis-codefresh** [comment]
The proper way is indeed notifications. And then you notify an second system (such as CI or other orchestrator) and that second system should rollback your external dependency.

Alternatively you can have your external system monitor the rollout itself and auto-rollback on its own.
